packages/react-aria-components/stories/animations.stories.tsx

```diff
@@ -0,0 +1,41 @@
+
+import {
+  Button,
+  Dialog,
+  DialogTrigger,
+  Heading,
+  Modal,
+  ModalOverlay
+} from 'react-aria-components';
+import React from 'react';
+import styles from './styles.css';
+
+
+export default {
+  title: 'React Aria Components - Animations'
+};
+
+export let ModalAnimation = {
```

Comment:
I'm not able to see the flicker with the original story. It fades in slow and it fades out slow.
In this version:
- it fades in slow completely and I close it with the button, it jumps to 3/4 closed and fades out.
- it fades in slow completely and I close it with escape it either completely closes or 7/8 closes and fades out.
- If I press escape as it's fading it, it jumps to fully open and starts fading out.

Reply:
I don't understand the comment. But you must let the animation play all the way through, there's no logic in place to have it do things like reverse mid-way